### PR TITLE
Orca: use build whl in pr-test

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Fix package source in PR-test and nightly-test

+ For a `PR-test`, need to first build whl then launch the test,e.g.
    ```bash
    bash python/dev/release_default_linux_spark3.sh default false false false -Ddata-store-url=$HTTP_URI -U
    pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
    ```
+ For `nightly-test`, should directly install packages from remote image, e.g. `pip install bigdl-orca-spark3`

Related issue: https://github.com/analytics-zoo/arda-docker/issues/773#issuecomment-1434153898

#### The following Orca PR tests:
- [x] Orca-Pytest-Jep-Py37-Spark3
- [x] Orca-Pytest-Ray-Ctx-Py37-Spark3
- [x] Orca-Pytest-Tf1-Py37-Spark3
- [x] Orca-Pytest-Openvino-Py37-Spark3
- [x] Orca-Pytest-Ray-Ctx-Py37-Spark3
- [x] Orca-Tutorial-Xshards-Py37-Spark3
- [x] Orca-Tutorial-Notebook-Xshards-Py37-Spark3
- [x] Orca-Tutorial-Notebook-Xshards-image-Py37-Spark3

#### ❗ The following nightly-test:
- [ ] Orca-Pytest-Tf1-Py37-Spark3
- [ ] Orca-ExampleTest-Jep-Py37-Spark3
- [x] Orca-ExampleTest-Ray-Ctx-Py37-Spark3
- [ ] Orca-ExampleTest-Ray-Py37-Spark3

### CODE REVIEW

File: .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml [hunk 1]
```diff
@@ -27,7 +27,16 @@ runs:
         conda create -n orca-tutorial-xshards-py37 -y python==3.7.10 setuptools==58.0.4 -c ${GONDOLIN_CONDA_CHANNEL} --override-channels
         conda info -e
         source activate orca-tutorial-xshards-py37
-        pip uninstall -y bigdl-friesian bigdl-friesian-spark3 bigdl-dllib bigdl-dllib-spark3 bigdl-orca pyspark bigdl-orca-spark3 bigdl-chronos bigdl-chronos-spark3 bigdl-friesian bigdl-friesian-spark3
```
Comment (hamham223) [inline .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml:30]: uninstalled in setup env, removed here

File: .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml [hunk 2]
```diff
@@ -27,7 +27,16 @@ runs:
...
+        
+        # install dllib
+        pip install -i https://pypi.org/simple python/dllib/src/dist/bigdl_dllib*-py3-none-manylinux1_x86_64.whl
+
+        # install orca
+        pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
+
         pip uninstall -y intel-tensorflow
         pip uninstall -y tensorflow
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} --pre --upgrade bigdl-orca-spark3[ray]
```
Comment (hamham223) [inline .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml:42]: Still need this line to install `requirements_ray.txt`?

File: .github/actions/orca/setup-env/setup-jep-prtest-py37/action.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        
+        # install dllib
+        pip install -i https://pypi.org/simple python/dllib/src/dist/bigdl_dllib*-py3-none-manylinux1_x86_64.whl
+        # install orca
+        pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
+
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} -r ${{ github.workspace }}/requirements-orca-python-jep.txt
+        pip uninstall -y opencv-python-headless
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} opencv-python-headless
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} fsspec==2021.9.0
```
Comment (hamham223) [inline .github/actions/orca/setup-env/setup-jep-prtest-py37/action.yml:41]: Can we now remove `fsspec`?

Comment (hkvision) [reply .github/actions/orca/setup-env/setup-jep-prtest-py37/action.yml:41]: I think yes

File: .github/workflows/PR_validation.yml [hunk 4]
```diff
@@ -469,6 +469,9 @@ jobs:
     - uses: actions/checkout@v3
     - name: Run Test
       uses: ./.github/actions/orca/orca-pytest-ray-ctx-py37-spark3-action
+    - name: Remove Env
```
Comment (hamham223) [inline .github/workflows/PR_validation.yml:472]: WIP: add setup env?

File: .github/actions/orca/orca-exampletest-ray-py37-spark3-action/nightly-test/action.yml [hunk 5]
```diff
@@ -29,7 +28,7 @@ runs:
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} matplotlib==3.5.3
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} pyarrow==4.0.1
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} albumentations
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} protobuf==3.15.3
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} protobuf==3.19.5
```
Comment (hkvision) [inline .github/actions/orca/orca-exampletest-ray-py37-spark3-action/nightly-test/action.yml:31]: don't need this, already included in [ray]

File: .github/actions/orca/orca-exampletest-ray-py37-spark3-action/nightly-test/action.yml [hunk 6]
```diff
@@ -29,7 +28,7 @@ runs:
...
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} fsspec==2021.9.0
```
Comment (hkvision) [inline .github/actions/orca/orca-exampletest-ray-py37-spark3-action/nightly-test/action.yml:32]: remove fsspec

File: .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml [hunk 7]
```diff
@@ -17,7 +17,7 @@ runs:
         # install orca
         pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
 
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} ray[default]==1.9.2
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} -r python/requirements/orca/requirements_ray.txt
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} openvino==2022.3.0
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} Pillow
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} protobuf==3.19.5
```
Comment (hkvision) [inline .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml:23]: remove protobuf

File: .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml [hunk 8]
```diff
@@ -17,7 +17,7 @@ runs:
         # install orca
         pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
 
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} ray[default]==1.9.2
```
Comment (hkvision) [inline .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml:20]: btw does openvino needs ray?

File: .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml [hunk 9]
```diff
@@ -27,17 +27,24 @@ runs:
...
         pip uninstall -y tensorflow
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} --pre --upgrade bigdl-orca-spark3[ray]
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow==1.15.0
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} scikit-learn==0.22.2.post1
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} pandas
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} jep==3.9.0
         pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} cloudpickle
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} ray[default]==1.9.2
-        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} protobuf==3.15.3
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} -r python/requirements/orca/requirements_ray.txt
```
Comment (hkvision) [inline .github/actions/orca/orca-tutorial-xshards-py37-spark3-action/action.yml:47]: always put this after installing orca

File: .github/actions/orca/setup-env/setup-jep-prtest-py37/action.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
+name: 'Run Orca Python Jep PR-test Py37'
```
Comment (hkvision) [inline .github/actions/orca/setup-env/setup-jep-prtest-py37/action.yml:1]: rename the folder to align with others, prtest is awkward.

File: .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml (hunk 8 above)
Comment (hamham223) [reply .github/actions/orca/orca-pytest-openvino-py37-spark3/action.yml:20]: Pillow may be also unnecessary 😏

File: .github/workflows/PR_validation.yml [hunk 11]
```diff
@@ -463,12 +463,17 @@ jobs:
   Orca-Pytest-Ray-Ctx-Py37-Spark3:
     needs: changes
     if: ${{ needs.changes.outputs.orca-pytest == 'true' }}
-    runs-on: [self-hosted, Gondolin-Ctx]
+    runs-on: [self-hosted, Gondolin-Ctx-Test]
```
Comment (hamham223) [inline .github/workflows/PR_validation.yml:466]: WIP: need to change the runner conf

**Comment by hamham223** [comment]
@liu-shaojun  Could you plz help me review the configurations (aka `action.yml`) for two ctx jobs? Especially for the example test. Thx!

**Pull request merged**